### ISSUE TITLE
Python 3.8 CLI raises ModuleNotFoundError: No module named 'typing_extensions'

### ISSUE THREAD

**Issue by gregbrowndev** [apache/airflow#9508]
**Apache Airflow version**: 1.10.10

**Environment**:

- **OS** (e.g. from /etc/os-release):

NAME=Fedora
VERSION="29 (Workstation Edition)"
ID=fedora
VERSION_ID=29
VERSION_CODENAME=""
PLATFORM_ID="platform:f29"
PRETTY_NAME="Fedora 29 (Workstation Edition)"
ANSI_COLOR="0;34"
LOGO=fedora-logo-icon
CPE_NAME="cpe:/o:fedoraproject:fedora:29"
HOME_URL="https://fedoraproject.org/"
DOCUMENTATION_URL="https://docs.fedoraproject.org/en-US/fedora/f29/system-administrators-guide/"
SUPPORT_URL="https://fedoraproject.org/wiki/Communicating_and_getting_help"
BUG_REPORT_URL="https://bugzilla.redhat.com/"
REDHAT_BUGZILLA_PRODUCT="Fedora"
REDHAT_BUGZILLA_PRODUCT_VERSION=29
REDHAT_SUPPORT_PRODUCT="Fedora"
REDHAT_SUPPORT_PRODUCT_VERSION=29
PRIVACY_POLICY_URL="https://fedoraproject.org/wiki/Legal:PrivacyPolicy"
VARIANT="Workstation Edition"
VARIANT_ID=workstation

- **Kernel** (e.g. `uname -a`):

Linux LAP300.itoworld.internal 4.20.6-200.fc29.x86_64 #1 SMP Thu Jan 31 15:50:43 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux

- **Install tools**: poetry

**What happened**:

Clean install of Airflow. Executing the `list_dags` CLI command throws an error:

```
$ python --version
Python 3.8.0
$ airflow list_dags
Traceback (most recent call last):
  File "/home/gbrown/.cache/pypoetry/virtualenvs/bods-ext-airflow-r4KRMLis-py3.8/bin/airflow", line 26, in <module>
    from airflow.bin.cli import CLIFactory
  File "/home/gbrown/.cache/pypoetry/virtualenvs/bods-ext-airflow-r4KRMLis-py3.8/lib/python3.8/site-packages/airflow/bin/cli.py", line 79, in <module>
    api_module = import_module(conf.get('cli', 'api_client'))  # type: Any
  File "/home/gbrown/.pyenv/versions/3.8.0/lib/python3.8/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "/home/gbrown/.cache/pypoetry/virtualenvs/bods-ext-airflow-r4KRMLis-py3.8/lib/python3.8/site-packages/airflow/api/client/local_client.py", line 24, in <module>
    from airflow.api.common.experimental import delete_dag
  File "/home/gbrown/.cache/pypoetry/virtualenvs/bods-ext-airflow-r4KRMLis-py3.8/lib/python3.8/site-packages/airflow/api/common/experimental/delete_dag.py", line 25, in <module>
    from airflow.models.serialized_dag import SerializedDagModel
  File "/home/gbrown/.cache/pypoetry/virtualenvs/bods-ext-airflow-r4KRMLis-py3.8/lib/python3.8/site-packages/airflow/models/serialized_dag.py", line 32, in <module>
    from airflow.serialization.serialized_objects import SerializedDAG
  File "/home/gbrown/.cache/pypoetry/virtualenvs/bods-ext-airflow-r4KRMLis-py3.8/lib/python3.8/site-packages/airflow/serialization/serialized_objects.py", line 37, in <module>
    from airflow.serialization.json_schema import Validator, load_dag_schema
  File "/home/gbrown/.cache/pypoetry/virtualenvs/bods-ext-airflow-r4KRMLis-py3.8/lib/python3.8/site-packages/airflow/serialization/json_schema.py", line 26, in <module>
    from typing_extensions import Protocol
ModuleNotFoundError: No module named 'typing_extensions'
```

Related issue: https://github.com/apache/airflow/pull/7474

**Comment by ipeluffo** [comment]
Strange, it seems this has been fixed in version 1.10.9:
* https://github.com/apache/airflow/pull/7474
* https://issues.apache.org/jira/browse/AIRFLOW-6854

And `master` doesn't have that import: https://github.com/apache/airflow/blob/d914a9c3f6bc900c93550675ee4bc71012a2e85e/airflow/serialization/json_schema.py#L21-L28

**Comment by kaxil** [comment]
This will be fixed in Airflow 1.10.11
